llvm/lib/Analysis/TargetLibraryInfo.cpp

```diff
@@ -882,11 +885,19 @@ static void initialize(TargetLibraryInfoImpl &TLI, const Triple &T,
   TLI.addVectorizableFunctionsFromVecLib(ClVectorLibrary, T);
 }
 
-TargetLibraryInfoImpl::TargetLibraryInfoImpl() {
-  // Default to everything being available.
-  memset(AvailableArray, -1, sizeof(AvailableArray));
```

Comment:
So we no longer default to everything being available? I'm not disagreeing with it because I think we need to select these more intelligently, but I'm wondering if that has an fallout.

Reply:
No, that's the next step I'm working towards

Reply:
> No, that's the next step I'm working towards

Not sure I follow - what is the next step? afaict this PR does what @jhuber6 was describing (no longer default to everything being available).

Reply:
The default constructed TLI should never be used. It shouldn't be possible to construct one without a triple, but there are places where one is default constructed and never used.

The ideal follow up change would be to not make anything available with an unknown target in the real constructor